libtenzir/builtins/contexts/geoip.cpp

```diff
@@ -531,9 +552,59 @@ struct v1_loader : public context_loader {
   }
 };
 
+struct v2_loader : public context_loader {
+  explicit v2_loader(record global_config)
+    : global_config_{std::move(global_config)} {
+  }
+
+  auto version() const -> int {
+    return 2;
+  }
+
+  auto load(chunk_ptr serialized) const
+    -> caf::expected<std::unique_ptr<context>> {
+    const auto* cache_dir
+      = get_if<std::string>(&global_config_, "tenzir.cache-directory");
+    std::filesystem::create_directories(*cache_dir);
+    const auto current_time = std::time(nullptr);
+    std::string temp_file_name = *cache_dir + std::to_string(current_time);
+    auto temp_file = std::fstream(temp_file_name, std::ios_base::out);
```

Comment:
Please use something like `{cache_dir}/plugins/geoip/{random_uuid}` for the filename.

Reply:
I second the request to move away from `std::time` for the intermediate file.
I'm not sure we should put it in `tenzir.cache-directory`, because looking at the `ctx` class I get the impression that the file gets read sometimes even after the context was created. A cache on the other hand should not be required for correctness.

Reply:
The logic has been changed so that the directory is purely a cache now. It never needs to be accessed later.